### PR TITLE
feature: replace existing signalering timestamp with latest

### PR DESCRIPTION
**Pull request by hsiliev**
Do not save a new signalering, but just update existing one's timestamp

Solves: PZ-2618

### CODE REVIEW

**Review by bas-info-nl — APPROVED** [review]
I was just checking if there's a simpler way to build json objects, and it seems that there is a [koson](https://github.com/lectra-tech/koson) library that could do that, worth checking out?

File: src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt [hunk 1]
```diff
@@ -125,11 +125,28 @@ class SignaleringService @Inject constructor(
...
     @Transactional(REQUIRED)
-    fun createSignalering(signalering: Signalering): Signalering {
+    fun storeSignalering(signalering: Signalering): Signalering {
         ValidationUtil.valideerObject(signalering)
-        val created = entityManager.merge(signalering)
-        eventingService.send(ScreenEventType.SIGNALERINGEN.updated(created))
-        return created
+
+        val signaleringToStore = findSignalering(signalering)?.apply {
+            LOG.warning("Replacing $this timestamp $tijdstip with ${signalering.tijdstip}")
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt:132]: I would make this loglevel INFO. It's normal behavior right? So not a warning I would say. We have INFO on on our test environment so we will see this come by.

**Review by edgarvonk — APPROVED** [review]
nice! small comment

**Comment by hsiliev** [comment]
> I was just checking if there's a simpler way to build json objects, and it seems that there is a [koson](https://github.com/lectra-tech/koson) library that could do that, worth checking out?

Are we talking about usage in tests (integration tests) here? Cause currently using string and `kotest` seems like a cool way to do this as we can simply copy-paste the json in our tests. 

We can however replace the json java library we now use as `koson` seems better in this regard.

File: src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt (hunk 1 above)
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt]: Done

File: src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt [hunk 2]
```diff
@@ -125,11 +125,28 @@ class SignaleringService @Inject constructor(
...
-        eventingService.send(ScreenEventType.SIGNALERINGEN.updated(created))
-        return created
+
+        val signaleringToStore = findSignalering(signalering)?.apply {
+            LOG.info("Replacing $this timestamp $tijdstip with ${signalering.tijdstip}")
+            tijdstip = signalering.tijdstip
+        } ?: signalering
+
+        return entityManager.merge(signaleringToStore).also {
+            eventingService.send(ScreenEventType.SIGNALERINGEN.updated(it))
```
Comment (hsiliev) [inline src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt:137]: In DB we do not have new signalering, but we still send screen event.

This means that in DB we have 1 event, while FE will list 2 with the real-time update. After refresh of the page the second one will disappear.

@infonl-marcel @edgarvonk I don't think this is correct. Perhaps we need to skip sending the event altogether?

Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt:137]: I don't know. Sounds like a functional question? Maybe make a proposal and ask Karin? What do users expect to happen?

Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt]: Moved the discussion to Slack

**Pull request merged**